Webhookoverview (#169)
elasticio/elasticio.github.io: content/_getting-started/webhooks-overview.md

---
title: Webhook Flow Overview
layout: article
section: Basic Concepts
order: 1
category: integration-flow
since: 20190406
---

This document provides basic information on [Webhooks](#webhooks) and [Webhook integration flows](#webhook-flows).

## Webhooks

> Webhooks allow applications to communicate between each other automatically. Here is a basic principle of their work:

Reviewer: ```suggestion
Webhooks allow applications to communicate with each other automatically. 

Let's say we have two applications, which are being run in different environments. And one of them, name it "sender application" have to send some data (or just notify) the other, name it "receiver application".

How this case is solved? Here is a typical approach:
```

> 1\. You set up a Webhook on the "receiver application", exposing a Webhook URL to receive data. Also, you configure this application to perform certain actions depending on the data it receives.

Reviewer: ```suggestion
1\. You set up the Webhook to the "receiver application". As a result, the "receiver application" got the HTTP(s) endpoint, where the other application could send data. The "receiver application" can process Webhooks' data whatever way it needs.
```

> 2\. You configure the "sender application" to send certain data to the "receiver application" Webhook URL as soon as a certain input triggers it. The data is usually a HTTP request in JSON format.

Reviewer: ```suggestion
2\. You configure the "sender application" to send certain data to the "receiver application" via sending an HTTP(s) request to the Webhook URL.
```

> 3\. The "action application" receives the data and acts accordingly, optionally replying with `404` if not successful, or some other pre-configured feedback.    

Reviewer: ```suggestion
3\. The "receiver application" receives the data and acts accordingly, optionally replying with `404` if not successful, or some other pre-configured feedback.
```

> Let's look at a small "real life" example for a clear vision. There is backend banking application, which actually moves the money and registers account states. Then, there is your personal client-bank application, typically web-based. This personal app is configured to receive transaction details on its exposed Webhook URL, and send you a report to your mobile messenger of choice. So as the backend app sends the appropriate data to client-bank Webhook URL, you immediately get a message about what is happening to your money.

Author: Let's have a live kurultai on this matter? I can understand "user-defined HTTP callbacks", you can understand them, but can all our potential readers?
